src/core/transpile.jl

```diff
@@ -86,7 +86,7 @@ end
 struct CompressSingleQubitGatesTranspiler<:Transpiler end
 
 # convert a single-target gate to a Universal gate
-function as_universal_gate(target::Integer,op::AbstractOperator)::Universal
+function as_universal_gate(target::Integer,op::AbstractOperator)::Gate
```

Comment:
as mentioned above, changing Gate to a generic struct could better constrain output type here

Reply:
Updated